internal/elasticsearch/client.go

```diff
@@ -369,3 +370,49 @@ func WaitForNumberOfHits(ctx context.Context, indexName string, query map[string
 	err := backoff.Retry(numberOfHits, exp)
 	return result, err
 }
+
+// GetSecurityApiKey waits for the elasticsearch SecurityApiKey to return the list of Api Keys.
+func GetSecurityApiKey() (*gabs.Container, error) {
+	//exp := utils.GetExponentialBackOff(60 * time.Second)
+
+	//retryCount := 1
+	//body := ""
+
+	//SecurityApiKeys := func()  error{
+	r := curl.HTTPRequest{
+		URL:               "http://localhost:9200/_security/api_key?",
+		BasicAuthPassword: "changeme",
+		BasicAuthUser:     "elastic",
+	}
+
+	response, err := curl.Get(r)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"error":          err,
+			"statusEndpoint": r.URL,
+		}).Warn("The Elasticsearch Cat Indices API is not available yet")
+
+		return nil, err
+	}
+
+	log.WithFields(log.Fields{
+		"statusEndpoint": r.URL,
+	}).Trace("The Elasticsearch Console SecurityApiKey API is available")
+
+	jsonParsed, err := gabs.ParseJSON([]byte(response))
```

Comment:
Would you mind adding the struct definition for what is expected to be return? I'd like to stay away from using gabs as much as possible so that we can continue to have better feedback during development.

Reply:
Yeah, I like this idea. Gabs is super flexible and helpful, but at the same time this flexibility comes with unstructured  responses that come adding difficulty when reading the code. A proper struct, as @adam-stokes suggests, would help in that readability

Reply:
I will work on it

Reply:
Done